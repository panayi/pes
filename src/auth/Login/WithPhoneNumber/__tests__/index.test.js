import React from 'react';
import { shallow } from 'enzyme';
import { WithPhoneNumber } from '../index';

describe('WithPhoneNumber', () => {
  beforeAll(() => {
    WithPhoneNumber.BUTTON_ID = 'mockButtonId';
  });

  it('should render correctly when showPhoneNumberForm is true', () => {
    const wrapper = shallow(<WithPhoneNumber showPhoneNumberForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when showCodeForm is true', () => {
    const wrapper = shallow(<WithPhoneNumber showCodeForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when showError is true', () => {
    const error = {
      message: 'something went wrong',
    };
    const wrapper = shallow(<WithPhoneNumber showError error={error} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when showLoading is true', () => {
    const wrapper = shallow(<WithPhoneNumber showLoading />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when showTryAgain is true', () => {
    const wrapper = shallow(<WithPhoneNumber showTryAgain />);
    expect(wrapper).toMatchSnapshot();
  });
});
