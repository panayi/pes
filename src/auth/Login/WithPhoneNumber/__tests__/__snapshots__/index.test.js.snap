// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithPhoneNumber should render correctly when showCodeForm is true 1`] = `
<div>
  <LocalForm
    initialState={Object {}}
    model="local"
    onSubmit={[Function]}
  >
    <DefaultConnectedControlText
      model=".code"
      placeholder="Enter SMS code"
    />
    <Styled(Styled(Base))
      id="mockButtonId"
      type="submit"
    >
      Submit
    </Styled(Styled(Base))>
  </LocalForm>
</div>
`;

exports[`WithPhoneNumber should render correctly when showError is true 1`] = `
<div>
  <span>
    something went wrong
  </span>
</div>
`;

exports[`WithPhoneNumber should render correctly when showLoading is true 1`] = `
<div>
  loading
</div>
`;

exports[`WithPhoneNumber should render correctly when showPhoneNumberForm is true 1`] = `
<div>
  <LocalForm
    initialState={Object {}}
    model="local"
    onSubmit={[Function]}
  >
    <DefaultConnectedControlText
      component={[Function]}
      defaultCountry="cy"
      model=".phoneNumber"
      placeholder="Enter phone number"
      validators={
        Object {
          "required": [Function],
        }
      }
    />
    <Styled(Styled(Base))
      id="mockButtonId"
      type="submit"
    >
      Sign in with phone number
    </Styled(Styled(Base))>
  </LocalForm>
</div>
`;

exports[`WithPhoneNumber should render correctly when showTryAgain is true 1`] = `
<div>
  <a
    onClick={[Function]}
    role="button"
    tabIndex="-1"
  >
    Try again
  </a>
</div>
`;
